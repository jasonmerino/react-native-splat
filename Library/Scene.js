import React, { PropTypes, Component } from 'react';
import {
  ScrollView,
  View,
} from 'react-native';
import KeyboardSpacer from 'react-native-keyboard-spacer';

class Scene extends Component {

  render() {
    return (
      <ScrollView
        {...this.props}
        style={[{
          flexGrow: 1,
        }, this.props.style]}
      >
        {this.props.children}
        <KeyboardSpacer />
      </ScrollView>
    );
  }

}

Scene.propTypes = {
  children: PropTypes.node,
  style: View.propTypes.style,
};

export default Scene;
