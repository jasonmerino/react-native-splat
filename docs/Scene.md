`Scene` (component)
===================



Props
-----

### `children`

type: `node`


### `style`

type: `custom`

