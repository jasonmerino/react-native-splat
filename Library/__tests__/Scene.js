
import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import Scene from '../Scene';

jest.mock('react-native-keyboard-spacer');

it('renders correctly', () => {
  const tree = renderer.create(
    <Scene />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
