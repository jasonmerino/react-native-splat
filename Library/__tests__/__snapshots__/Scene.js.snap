exports[`test renders correctly 1`] = `
<ScrollView
  style={
    Array [
      Object {
        "flexGrow": 1,
      },
      undefined,
    ]
  }>
  <KeyboardSpacer />
</ScrollView>
`;
